flagsmith/flagsmith.py

```diff
@@ -190,14 +204,16 @@ def get_identity_segments(
         segment_models = get_identity_segments(self._environment, identity_model)
         return [Segment(id=sm.id, name=sm.name) for sm in segment_models]
 
-    def update_environment(self):
+    def update_environment(self) -> None:
         self._environment = self._get_environment_from_api()
 
     def _get_environment_from_api(self) -> EnvironmentModel:
         environment_data = self._get_json_response(self.environment_url, method="GET")
         return EnvironmentModel.model_validate(environment_data)
 
     def _get_environment_flags_from_document(self) -> Flags:
+        if not is_environment_model(self._environment):
+            raise TypeError("Environment should not be 'None'")
```

Comment:
Nit: `self._environment` is probably `None` but it would be better to return the type explicitly like this:
```python
raise TypeError(f"Environment should not be '{type(self._environment)}'")
```

Reply:
Yes, agreed. Thanks!

Reply:
I'll take a closer look later this week but I'm not sure this is at all needed? A simple check for `None` should suffice here. This code is probably a leftover (or just bad Python code).

Reply:
Sure, I believe `None` is better here. `TypeGuards` now seems much more suitable when there are different types involved 🤔